src/routes/private.ts

```diff
@@ -8,14 +8,16 @@ import {
 } from '../entity';
 import { getConnection } from 'typeorm';
 import { SubmissionFailErrorKeys, SubmissionFailErrorMessage } from '../errors';
+import pino from 'pino';
 
 export default async function (fastify: FastifyInstance): Promise<void> {
   fastify.post<{ Body: AddPostData }>('/newPost', async (req, res) => {
     if (!req.service) {
       return res.status(404).send();
     }
     const con = getConnection();
-    const operationResult = await addNewPost(con, req.body);
+    const logger = pino();
+    const operationResult = await addNewPost(con, req.body, logger);
```

Comment:
The request object should already have a logger defined. No need to define a new instance.
In most cases you don't have to define an instance as a rule of thumb

Reply:
Updated to use the logger from the req object.